src/test/java/edu/harvard/iq/dataverse/engine/command/impl/RestrictFileCommandTest.java

```diff
@@ -89,19 +91,33 @@ public void testRestrictUnpublishedFile() throws CommandException{
     
     @Test
     public void testRestrictPublishedFile() throws Exception{
-        file.setOwner(dataset);
         dataset.setPublicationDate(new Timestamp(new Date().getTime()));
+        // Restrict on a published file will cause the creation of a new draft dataset version
+        // and should update only the FileMetadata in the draft version for the test file.
+        // So we need to make sure that we use one of the files in the dataset for the test 
+        DataFile file = dataset.getFiles().get(0);
+        // And make sure is is file.isReleased() == true
+        file.setPublicationDate(dataset.getPublicationDate());
+        // And set its owner, which is usually done automatically, but not in the test setup
+        file.setOwner(dataset);
+        //And set the version state to released so that the RestrictFileCommand will create a draft version
+        dataset.getLatestVersion().setVersionState(VersionState.RELEASED);
```

Comment:
All these updates do a better job of putting the dataset/it's initial version into the released state and having the file being restricted as one of the ones in the list of files in the dataset (and its metadata in the list of file metadatas).